cpp/src/parquet.cpp

```diff
@@ -250,6 +250,9 @@ class ParquetRead : public Task<ParquetRead, OpCode::ParquetRead> {
     auto opt = cudf::io::parquet_reader_options::builder(src);
     opt.columns(columns);
     opt.row_groups(row_groups);
+    // If pandas metadata is read, libcudf may read index columns without this.
+    opt.use_pandas_metadata(false);
```

Comment:
Ran into this with a weirder pandas generated parquet file, would need to include the parquet file to actually test (since I don't want to generate weird pandas metadata).